File: 2256-minimum-average-difference/2256-minimum-average-difference.py
class Solution:
    def minimumAverageDifference(self, nums: List[int]) -> int:
        n = len(nums)
        pre = [0] * (n + 1)
        
        for i in range(n):
            pre[i + 1] = pre[i] + nums[i]
        
        diff = inf
        answ = 0
        
        for i in range(1, n + 1):
            j = n - i
            a = int(pre[i] / i)
            b = 0 if not j else int((pre[n] - pre[i]) / j)
            t = abs(a - b)
            if t < diff:
                diff = t
                answ = i - 1
                
        return answ

File: 2256-minimum-average-difference/2256-minimum-average-difference.cpp
class Solution {
public:
    int minimumAverageDifference(vector<int>& nums) {
        using ll = long long;
        int n = nums.size();
        vector<ll> pre(n + 1, 0);
        for (int i = 0; i < n; ++i) {
            pre[i + 1] = pre[i] + 1LL*nums[i];
        }
        int diff = INT_MAX;
        int answ = 0;
        for (int i = 1; i <= n; ++i) {
            int j = n - i;
            int a = pre[i] / i;
            int b = !j ? 0 : (pre[n] - pre[i]) / j;
            int t = abs(a - b);
            if (diff > t) {
                diff = t;
                answ = i - 1;
            }
        }
        return answ;
    }
};